src/libasr/pass/global_stmts.cpp

```diff
@@ -141,7 +82,47 @@ void pass_wrap_global_stmts(Allocator &al,
 
     if (return_var) {
         // The last defined `return_var` is the actual return value
+        LCOMPILERS_ASSERT(type)
+        LCOMPILERS_ASSERT(return_var_ref)
+
         ASR::down_cast2<ASR::Variable_t>(return_var)->m_intent = ASRUtils::intent_return_var;
+        std::string global_underscore_name = "_" + fn_name_s;
+        s.from_str(al, global_underscore_name);
+        
+        ASR::asr_t *global_underscore = ASR::make_Variable_t(al, loc,
```

Comment:
Are we supporting `lists` in this PR or the `_` variable? If we are adding support for both, can you please break this into two PR with adding tests for each in the respective PRs?